### PR TITLE
Tableau connector api

### PR DESCRIPTION
**Pull request by LtChae**
Taking suggestions for the router paths

### CODE REVIEW

File: apps/discovery_api/lib/discovery_api/data/table_info_cache.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def invalidate() do
+    Cachex.clear(cache_name())
+  end
+
+  defp cache_name() do
+    :table_info
+  end
+end
```
Comment (mcsearchin) [inline apps/discovery_api/lib/discovery_api/data/table_info_cache.ex]: Not having looked at the other cache(s), would it make sense to consolidate any of these cachex boilerplate into a module that can be extended?

**Comment by kcarra** [comment]
I'm cool with the router path names :sungla

File: apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex [hunk 2]
```diff
@@ -51,6 +53,41 @@ defmodule DiscoveryApiWeb.MultipleMetadataController do
...
+        filtered_models ->
+          filtered_models
+      end
+
+    render(
+      conn,
+      :fetch_table_info,
+      models: filtered_models
+    )
+  end
```
Comment (mcsearchin) [inline apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex]: Is there any merit to the idea of putting tableau-related stuff into its own controller?  Not saying you need to make a bunch of changes - just a thought.

File: apps/discovery_api/lib/discovery_api_web/views/multiple_metadata_view.ex [hunk 3]
```diff
@@ -31,6 +31,10 @@ defmodule DiscoveryApiWeb.MultipleMetadataView do
     }
   end
 
+  def render("fetch_table_info.json", %{models: models}) do
+    models
+  end
```
Comment (mcsearchin) [inline apps/discovery_api/lib/discovery_api_web/views/multiple_metadata_view.ex]: Should this be `%{table_infos: table_infos}` - just to alleviate confusion?

File: apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex [hunk 4]
```diff
@@ -51,6 +53,41 @@ defmodule DiscoveryApiWeb.MultipleMetadataController do
...
 
+  def fetch_table_info(conn, _params) do
+    user_id = get_user_id(conn)
+
+    filtered_models =
+      case TableInfoCache.get(user_id) do
+        nil ->
+          remove_unauthorized_models(conn, Model.get_all())
+          |> get_filtered_table_info()
+          |> TableInfoCache.put(user_id)
```
Comment (mcsearchin) [inline apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex]: So, are we just caching with `nil` as the key if the user is anonymous?

Comment (LtChae) [reply apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex]: Yup

File: apps/discovery_api/lib/discovery_api/data/table_info_cache.ex (hunk 1 above)
Comment (kcarra) [reply apps/discovery_api/lib/discovery_api/data/table_info_cache.ex]: @mcsearchin I know that system_name_cache does something very similar to this.

File: apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex (hunk 2 above)
Comment (cameronmarsh) [reply apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex]: done

File: apps/discovery_api/lib/discovery_api/data/table_info_cache.ex (hunk 1 above)
Comment (cameronmarsh) [reply apps/discovery_api/lib/discovery_api/data/table_info_cache.ex]: Good idea. We decided to punt on this though, maybe we should write a card for it?

**Pull request merged**